ElementX/Sources/Screens/RoomScreen/View/Style/TimelineItemBubbledStylerView.swift

```diff
@@ -108,27 +112,50 @@ struct TimelineItemBubbledStylerView<Content: View, DeliveryStatus: View>: View
                              cornerRadius: cornerRadius,
                              corners: roundedCorners)
         } else {
-            VStack(alignment: .trailing, spacing: 4) {
-                contentWithReply
+            contentWithTimestamp
+                .bubbleStyle(inset: true,
+                             color: timelineItem.isOutgoing ? .element.bubblesYou : .element.bubblesNotYou,
+                             cornerRadius: cornerRadius,
+                             corners: roundedCorners)
+        }
+    }
 
-                if timelineItem.properties.isEdited {
-                    Text(L10n.commonEditedSuffix)
-                        .font(.compound.bodyXS)
-                        .foregroundColor(.element.tertiaryContent)
+    @ViewBuilder
+    var contentWithTimestamp: some View {
+        if isTextItem {
+            ZStack(alignment: .topLeading) {
+                contentWithReply
+                    .layoutPriority(1)
+                VStack(alignment: .trailing, spacing: 0) {
+                    Spacer()
+                    HStack(alignment: .bottom, spacing: 0) {
+                        Spacer()
+                        timestamp
+                    }
                 }
+            }
+        } else {
+            HStack(alignment: .bottom, spacing: 4) {
+                contentWithReply
+                timestamp
+            }
+        }
+    }
 
-                if timelineItem.properties.deliveryStatus == .sendingFailed {
-                    Image(systemName: "exclamationmark.circle.fill")
-                        .resizable()
-                        .foregroundColor(.element.alert)
-                        .frame(width: 16, height: 16)
-                }
+    @ViewBuilder
+    var timestamp: some View {
+        HStack(spacing: 4) {
+            Text(timelineItem.timestampString)
+                .font(.compound.bodyXS)
+                .foregroundColor(timelineItem.properties.deliveryStatus == .sendingFailed ? .element.alert : .element.secondaryContent)
+            if timelineItem.properties.deliveryStatus == .sendingFailed {
+                Image(systemName: "exclamationmark.circle.fill")
+                    .resizable()
+                    .foregroundColor(.element.alert)
+                    .frame(width: 12, height: 12)
```

Comment:
Not sure here, but do we want this to scale with DynamicType to match the rest of the string? If so using the same font works perfectly here:
```suggestion
                    .font(.compound.bodyXS)
                    .foregroundColor(.element.alert)
```
And in-fact, you could now extract the font and colour up one level to the HStack (or even inline the `Image` into the `Text` and remove the stack entirely) to only use the modifiers once.